src/custom/state/appData/updater.ts

```diff
@@ -78,8 +77,31 @@ async function _uploadToIpfs(
       }
     } catch (e) {
       // TODO: add sentry error to track soft failure
-      console.debug(`[UploadToIpfsUpdater] Failed to upload doc, will try again`, JSON.stringify(doc))
-      updatePending({ chainId, orderId, uploading: false, tryAfter: Date.now() + TRY_AGAIN_AFTER })
+      console.debug(
+        `[UploadToIpfsUpdater] Failed to upload doc, will try again. Reason: ${e.message}`,
+        JSON.stringify(doc)
+      )
+      updatePending({ chainId, orderId, uploading: false, failedAttempts: failedAttempts + 1, lastAttempt: Date.now() })
     }
+  } else {
```

Comment:
ubernit, this else is many lines away of the condition. In other words, maybe the branches of this if/else can be moved partially or completly to auxiliary functions, so the whole thing is easier to read?

Reply:
To be honest this bit was added only for debugging and can be removed.

Anyway, will refactor